frontend/src/components/modules/Login/Login.tsx

```diff
@@ -14,88 +22,174 @@ interface FormData {
   password: string;
 }
 
+interface ErrorMessage {
+  message: string;
+}
+
 const LoginForm: React.FC = () => {
   const navigate = useNavigate();
   const { login } = useAuth();
   const location = useLocation() as unknown as LocationProps;
   const from = location.state?.from?.pathname ?? "/";
+  const loginAPI = "http://localhost:8080/api/auth/login"; // transfer the base URL into env file.
 
   const [formData, setFormData] = useState<FormData>({
     login: "",
     password: ""
   });
 
-  const onHandleSubmit = async (): Promise<void> => {
-    await login();
-    navigate(from, { replace: true });
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage>({
+    message: ""
+  });
+
+  const onHandleSubmit = async (
+    event: React.FormEvent<HTMLFormElement>
+  ): Promise<void> => {
+    event.preventDefault(); // Prevent the default form submit behavior
+    try {
+      const errorContext = ["requestBody.password", "requestBody.login"];
+      const response = await fetch(loginAPI, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json"
+        },
+        body: JSON.stringify({
+          email: formData.login,
+          password: formData.password
+        })
+      });
+
+      if (response.ok) {
+        const data = await response.json();
+        console.log(data);
+        // Store the token, and update auth state
+        await login();
+        navigate(from, { replace: true });
+      } else {
+        // TODO: display a message to the user
+        const errorData = await response.json();
+        const context = errorData.errors[0].context;
+        for (let i = 0; i < errorContext.length; i++) {
+          const key = errorContext[i];
+          if (context[key] !== undefined) {
+            setErrorMessage({
+              message: context[key].message
+            });
+            break; // found the error message
+          }
```

Comment:
Same thing here. Maybe we could come up with some reusable error-handling utility function that could be called everywhere else as well considering that we will most likely have multiple places where these errors need to be handled.

Reply:
I can use a common module for error handling maybe. And import it in where its needed.

Reply:
done @vilbedawg

Reply:
I meant that the parsing of the context value should be moved inside the error component as well, rather than writing this logic on every component that want's to show error messages.

Also, there can be more than a single context. If validation fails, it will display all the invalid fields, so breaking out of the loop after finding the context shouldn't happen. Also I'm not actually sure if the context is something we want to show to the user.

Reply:
I will create a separate PR for error management. I have already worked on the registration API call. I will figure out how to manage the error component most efficiently.